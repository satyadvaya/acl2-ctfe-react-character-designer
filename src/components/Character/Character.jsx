import React from "react";

export default function Character({ head, middle }) {
  return (
    <section style={{ display: "flex", flexDirection: "column" }}>
      <article
        className="head"
        style={{
          height: 150,
          width: 150,
          backgroundImage: `url(./${head}-head.png)`,
          backgroundSize: "cover",
          backgroundRepeat: "no-repeat",
        }}
      ></article>

      <article
        className="middle"
        style={{
          height: 150,
          width: 150,
          backgroundImage: `url(./${middle}-middle.png)`,
          backgroundSize: "cover",
          backgroundRepeat: "no-repeat",
        }}
      ></article>
    </section>
  );
}
