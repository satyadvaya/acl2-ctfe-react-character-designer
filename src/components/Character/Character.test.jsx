import { screen, render } from "@testing-library/react";
import Character from "./Character";

it("should render the Charater component", () => {
  const { container } = render(
    <Character head="bird" middle="fancy" pants="blue" />
  );

  // const head = screen.getByLabelText(/head/);
  // const middle = screen.getByLabelText(/middle/);
  // const pants = screen.getByLabelText(/pants/);

  // expect(head).toHaveStyle(`background-image: url(./bird-head.png)`);
  // expect(middle).toHaveStyle(`background-image: url(./fancy-middle.png)`);
  // expect(pants).toHaveStyle(`background-image: url(./blue-pants.png)`);

  expect(container).toMatchSnapshot();
});
