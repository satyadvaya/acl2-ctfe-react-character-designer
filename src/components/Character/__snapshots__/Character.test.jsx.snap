// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Character component 1`] = `
<div>
  <section
    style="display: flex; flex-direction: column;"
  >
    <article
      class="head"
      style="height: 150px; width: 150px; background-image: url(./bird-head.png); background-size: cover; background-repeat: no-repeat;"
    />
    <article
      class="middle"
      style="height: 150px; width: 150px; background-image: url(./fancy-middle.png); background-size: cover; background-repeat: no-repeat;"
    />
    <article
      class="pants"
      style="height: 150px; width: 150px; background-image: url(./blue-pants.png); background-size: cover; background-repeat: no-repeat;"
    />
  </section>
</div>
`;
